### PR TITLE
8263486: Clean up MTLSurfaceDataBase.h

### PR DESCRIPTION
**Pull request by aghaisas**
This PR addresses some cleanup activities :
- Cleaned up MTLSurfaceDataBase.h & MTLSurfaceData.m
- Removed OpenGL references from MTLPipelineStatesStorage.m  & MTLRenderQueue.m

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263486](https://bugs.openjdk.java.net/browse/JDK-8263486): Clean up MTLSurfaceDataBase.h

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4010/head:pull/4010` \
`$ git checkout pull/4010`

Update a local copy of the PR: \
`$ git checkout pull/4010` \
`$ git pull https://git.openjdk.java.net/jdk pull/4010/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4010`

View PR using the GUI difftool: \
`$ git pr show -t 4010`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4010.diff">https://git.openjdk.java.net/jdk/pull/4010.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceDataBase.h [hunk 1]
```diff
@@ -48,43 +47,33 @@
...
- * Windows XP has lower-left insets of (4,4).  The Metal viewport origin
- * would typically begin at the lower-left corner of the client region (inside
- * the frame decorations), but AWT/Swing will take the insets into account
- * when rendering into that window.  So in order to account for this, we
- * need to adjust the Metal viewport origin by an x/yOffset of (-4,-4).  On
- * X11, top-level frames typically don't have this insets issue, so their
- * x/yOffset would be (0,0) (the same applies to pbuffers).
+ *     jint x/yOffset;
+ * The offset in pixels of the viewport origin from the lower-left
+ * corner of the heavyweight drawable.
```
Comment (mrserb) [inline src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceDataBase.h:56]: Do we use these fields or they are always zero?

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceData.m [hunk 2]
```diff
@@ -28,20 +28,11 @@
...
 
-/**
- * The following methods are implemented in the windowing system (i.e. GLX
- * and WGL) source files.
- */
-extern jlong MTLSD_GetNativeConfigInfo(BMTLSDOps *bmtlsdo);
-extern jboolean MTLSD_InitMTLWindow(JNIEnv *env, BMTLSDOps *bmtlsdo);
-extern void MTLSD_DestroyMTLSurface(JNIEnv *env, BMTLSDOps *bmtlsdo);
-
+jboolean MTLSD_InitMTLWindow(JNIEnv *env, BMTLSDOps *bmtlsdo);
```
Comment (mrserb) [inline src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceData.m:35]: How the MTLSD_WINDOW is used? Do we use it for the layer-based rendering?

Comment (aghaisas) [reply src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceData.m:35]: It is used in MTLContex.m. It just indicates a surface that we render to. We blit from this surface to the CAMetalLayer lateron.
It looks like MTLSD_WINDOW can be replaced with simple MTLSD_TEXTURE. This investigation and cleanup has already been identified during lanai code review (Bug - JDK-8263463)

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**